src/lazy/encoder/binary/v1_1/container_writers.rs

```diff
@@ -22,23 +21,90 @@ use crate::{FlexUInt, IonResult, RawSymbolTokenRef, SymbolId};
 pub(crate) struct BinaryContainerWriter_1_1<'value, 'top> {
     // An allocator reference that can be shared with nested container writers
     allocator: &'top BumpAllocator,
-    // The buffer to which child values will be encoded. In the case of:
-    //   1. a length-prefixed container, this will be a new buffer bump-allocated specifically for this
-    //      container.
-    //   2. a delimited container, this will be the parent's own encoding buffer, to which the delimited
-    //      container start opcode has already been written.
+    encoder: ContainerEncodingKind<'value, 'top>,
+}
+
+enum ContainerEncodingKind<'value, 'top> {
+    Delimited(DelimitedEncoder<'value, 'top>),
+    LengthPrefixed(LengthPrefixedEncoder<'value, 'top>),
+}
```

Comment:
🗺️ To enable containers to be either delimited or length prefixed, `BinaryContainerWriter_1_1` now holds an enum of the required encoding resources instead of a pointer to a buffer.